mm/2s2h/Enhancements/DifficultyOptions/GibdoTradeSequenceOptions.cpp

```diff
@@ -0,0 +1,72 @@
+#include <libultraship/bridge.h>
+#include "2s2h/GameInteractor/GameInteractor.h"
+#include "2s2h/Enhancements/Enhancements.h"
+#include "2s2h/ShipInit.hpp"
+#include <stdarg.h>
+
+#define CVAR_NAME "gEnhancements.Cheats.GibdoTradeSequence"
+#define CVAR CVarGetInteger(CVAR_NAME, GIBDO_TRADE_SEQUENCE_VANILLA)
+
+extern "C" {
+#include "functions.h"
+#include "variables.h"
+#include "overlays/actors/ovl_En_Talk_Gibud/z_en_talk_gibud.h"
+
+// redefinition
+typedef struct {
+    /* 0x0 */ PlayerItemAction itemAction;
+    /* 0x4 */ ItemId item;
+    /* 0x8 */ s32 amount;
+    /* 0xC */ s16 isBottledItem;
+} EnTalkGibudRequestedItem; // size = 0x10
+}
+
+static EnTalkGibudRequestedItem redPotionRequestedItem = { PLAYER_IA_BOTTLE_POTION_RED, ITEM_POTION_RED, 1, true };
+
+void RegisterGibdoTradeSequenceOptions() {
+    COND_VB_SHOULD(VB_GIBDO_TRADE_SEQUENCE_SUFFICIENT_QUANTITY_PRESENTED, CVAR != GIBDO_TRADE_SEQUENCE_VANILLA, {
+        ItemId requestedItemId = va_arg(args, ItemId);
+        if (AMMO(requestedItemId) >= 1) {
+            *should = true;
+        }
+    });
+
+    COND_VB_SHOULD(VB_GIBDO_TRADE_SEQUENCE_ACCEPT_RED_POTION, CVAR == GIBDO_TRADE_SEQUENCE_MM3D, {
+        PlayerItemAction requestedItemAction = va_arg(args, PlayerItemAction);
+        PlayerItemAction presentedItemAction = va_arg(args, PlayerItemAction);
```

Comment:
I believe this is required when passing things like enums through va_args, someone can correct me if I'm wrong
```suggestion
        PlayerItemAction requestedItemAction = (PlayerItemAction)va_arg(args, int);
        PlayerItemAction presentedItemAction = (PlayerItemAction)va_arg(args, int);
```

Reply:
This is correct, see #840 for example